### ISSUE TITLE
Automated initialization method for Costing Blocks

### ISSUE THREAD

**Issue by andrewlee94** [IDAES/idaes-pse#746]
As part of the improvements to the IDAES Costing Framework, we realized that for most costing Blocks it should be possible to automate initialization by doing a block-triangularization of the Block in order to determine the initialization order and to then sequentially solve the constraints in order.

**Comment by andrewlee94** [comment]
@Robbybp Just a reminder of this - I am actually wondering if we should try to make this a global tool (as a default option for all models).

**Comment by Robbybp** [comment]
Yes, I would like to try and implement something general for initialization via a block-triangularization. Ideally this would be faster than current initialization, but possibly less reliable for certain models/definitions of reliability (although possibly more reliable). I think it would be nice to use the performance suite to benchmark initialization times with such a strategy.
